examples/at_uart/main.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief  at_uart
+ *
+ * @author  RocioRojas <rociorojas391@gmail.com>
+ * @author  eduazocar <eduardo@turpialdev.com>
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "board.h"
+#include "msg.h"
+#include "shell.h"
+#include "shell_commands.h"
+#include "thread.h"
+#include "periph/uart.h"
+#include "xtimer.h"
+
+#define AT_UART UART_DEV(1)
+#define AT_BAUDRATE 115200
+#define MAX_SENTENCE_SIZE 128
+
+#define MAIN_QUEUE_SIZE (8)
+msg_t _main_msg_queue[MAIN_QUEUE_SIZE];
+
+typedef struct t_at_list {
+    char *shell_cmd;
+    char *value_desc;
+    char *at_cmd;
+    uint8_t arg;
+} tAT_LIST;
+
+enum AT_COUNT {
+    WAP_SSID = 0,
+    WAP_PASS,
+    WSTA_SSID,
+    WSTA_PASS,
+    WIFI_OFF,
+    WIFI_ON,
+    NVS_RST,
+    WIFI_MODE,
+    WAP_CHAN,
+    WAP_AUTH,
+    WIFI_RST,
+    MAX_AT,
+};
+
+tAT_LIST at_list[] = {
+    // shell command, value description, at command, argument
+    {"wap_ssid", "<AP SSID>", "AT+WAP_SSID=", 1},
+    {"wap_pass", "<AP Password>", "AT+WAP_PASS=", 1},
+    {"wsta_ssid", "<STA SSID>", "AT+WSTA_SSID=", 1},
+    {"wsta_pass", "<STA Password>", "AT+WSTA_PASS=", 1},
+    {"woff", "", "AT&WIFI_OFF", 0},
+    {"won", "", "AT&WIFI_ON", 0},
+    {"nvs_rst", "", "AT&NVS_RST", 0},
+    {"wmode", "<0 / 1 / 2 / 3>", "AT+WIFI_MODE=", 1},
+    {"wap_chan", "<Number>", "AT+WAP_CHAN=", 1},
+    {"wap_auth", "<Number as AUTH_TYPE>", "AT+WAP_AUTH=", 1},
+    {"wrst", "", "AT&WIFI_RST\r\n", 0},
+    {NULL, NULL, NULL, 0},
+};
+
+size_t count = 0;
+char uart_buff[MAX_SENTENCE_SIZE] = {0};
+
+static void uart_rx_cb(void *arg, uint8_t c) {
+    printf("%s\n", (char *)arg);
+    if (c != 10) {
+        uart_buff[count] = (char)c;
+        count += 1;
+    } else {
+        printf("%s\n", (char *)uart_buff);
+        memset(uart_buff, 0, sizeof(uart_buff));
+        count = 0;
+    }
+}
```

Comment:
this is taken from an older example, remove it or adjust it for your requirements